tests/distributed/test_distributed_sampling.py

```diff
@@ -417,6 +406,49 @@ def check_rpc_hetero_sampling_shuffle(tmpdir, num_server):
         assert np.all(F.asnumpy(orig_src1) == orig_src)
         assert np.all(F.asnumpy(orig_dst1) == orig_dst)
 
+def get_degrees(g, nids, ntype):
+    deg = F.zeros((len(nids),), dtype=F.int64)
+    for srctype, etype, dsttype in g.canonical_etypes:
+        if srctype == ntype:
+            deg += g.out_degrees(u=nids, etype=etype)
+        elif dsttype == ntype:
+            deg += g.in_degrees(v=nids, etype=etype)
+    return deg
+
+def check_rpc_hetero_sampling_empty_shuffle(tmpdir, num_server):
+    ip_config = open("rpc_ip_config.txt", "w")
+    for _ in range(num_server):
+        ip_config.write('{}\n'.format(get_local_usable_addr()))
+    ip_config.close()
+
+    g = create_random_hetero(empty=True)
+    num_parts = num_server
+    num_hops = 1
+
+    orig_nids, _ = partition_graph(g, 'test_sampling', num_parts, tmpdir,
+                                   num_hops=num_hops, part_method='metis',
+                                   reshuffle=True, return_mapping=True)
+
+    pserver_list = []
+    ctx = mp.get_context('spawn')
+    for i in range(num_server):
+        p = ctx.Process(target=start_server, args=(i, tmpdir, num_server > 1, 'test_sampling'))
+        p.start()
+        time.sleep(1)
+        pserver_list.append(p)
+
+    time.sleep(3)
+    deg = get_degrees(g, orig_nids['n3'], 'n3')
```

Comment:
We can use remove edges to create isolated nodes.
Anyway, current solution also works.

Reply:
how to remove edges? we can also create an empty graph, but i don't know how graph partition works.